NewEmuChecklist.txt

```diff
@@ -0,0 +1,75 @@
+RetroAchievements Emulator Requirements:
+* If the emulator does not support a feature, the requirement is automatically met. This is primarily a set of
+  guidelines to ensure the interaction between the emulator and the toolkit is consistent between emulators.
+* If the system is supported by RetroArch, explain why a standalone emulator is desirable. RetroArch support is 
+  preferred as it is usable by more players (i.e. not just Windows users), and adding support to RALibRetro for 
+  an additional core is much easier than overhauling an entire emulator.
+
+[ ] Game Identification
+  [ ] Hash method documented and approved
+  [ ] If emulator supports dynamic patching [i.e. ips], hash is calculated after applying patch
+  [ ] Hash ignores any persistant save data [SRAM]
+  [ ] Game can be properly identified (RA_OnLoadNewRom)
+  [ ] Unidentified game provides some descriptor [typically filename] to unknown game dialog (RA_InstallSharedFunctions>RA_GetEstimatedGameTitle)
+  [ ] Unidentified game provides list of known games for console (RA_SetConsoleID)
+[ ] Emulator
+  [ ] DLL is downloaded if not available (RA_Init - new enum may be needed)
+  [ ] RetroAchievements menu is visible (RA_CreatePopupMenu) and functional (RA_InvokeDialog)
+  [ ] RetroAchievements menu updates when logging in/out (RA_InstallSharedFunctions>RA_RebuildMenu)
+  [ ] Login dialog displayed on first run or auto-login on future runs (RA_AttemptLogin)
+  [ ] Title includes logged in user (RA_UpdateAppTitle)
+  [ ] RALog.txt ends with "Shutdown complete" after closing emulator (RA_Shutdown)
+  [ ] Full-window overlay can be displayed (RA_SetPaused)
+  [ ] Full-window overlay can be navigated (RA_UpdateRenderOverlay>pInput)
+  [ ] Full-window overlay navigation uses player's custom controller bindings, not hard-coded inputs.
+  [ ] Closing the full-window overlay unpauses the emulator (RA_InstallSharedFunctions>RA_CauseUnpause)
+  [ ] Resetting the emulator disables any active popups/leaderboards (RA_OnReset)
+  [ ] Performance of emulator (without any debugging windows open) should not be noticably different than the source emulator
+  [ ] Emulator must perform at intended framerate on an "average" machine.
```

Comment:
Defining "average" might be helpful.

Reply:
It would, but I don't have that metric. It's something GameDragon requested I add. Apparently some of the attempted previous emulators ran like crap even before trying to integration RA.

Reply:
I've decided to say "a PC that could have been purchased at a retail store within the last four years". It's still pretty subjective as there's a big difference between what Walmart sells and what Best Buy sells.